desc/objectives/_qs.py

```diff
@@ -508,3 +508,136 @@ def compute(self, *args, **kwargs):
             profiles=self._profiles,
         )
         return data["f_T"] * self.grid.weights
+
+
+class Isodynamicity(_Objective):
+    """Isodynamicity metric for cross field transport.
+
+    Note: This is NOT the same as Quasi-isodynamicity (QI), which is a more general
+    condition. This specifically penalizes the local cross field transport, rather than
+    just the average.
+
+    Parameters
+    ----------
+    eq : Equilibrium, optional
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : float, ndarray, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        len(target) must be equal to Objective.dim_f
+    bounds : tuple, optional
+        Lower and upper bounds on the objective. Overrides target.
+        len(bounds[0]) and len(bounds[1]) must be equal to Objective.dim_f
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(weight) must be equal to Objective.dim_f
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+        Has no effect for this objective.
+    normalize_target : bool
+       Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+        Has no effect for this objective.
+    grid : Grid, ndarray, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = False
+    _units = "(~)"
+    _print_value_fmt = "Isodynamicity error: {:10.3e} "
+
+    def __init__(
+        self,
+        eq=None,
+        target=0,
+        bounds=None,
+        weight=1,
+        normalize=False,
+        normalize_target=False,
```

Comment:
The default for all of our dimensionless objectives is `normalize=False`, like we have here. That is fine for scalar objectives (like aspect ratio, etc.) where the normalization doesn't do anything. But for non-scalar objectives like this one, it could still be useful to normalize by the number of collocation points: `self._normalization = 1 / jnp.sqrt(self._dim_f)`